### PR TITLE
Ledgerstate Refactor: Address

### PR DESCRIPTION
**Pull request by hmoog**
# Description of change

This PR introduces the refactored models used by the addresses.

## Change checklist

Add an `x` to the boxes that are relevant to your changes, and delete any items that are not.

- [x] My code follows the contribution guidelines for this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes

### CODE REVIEW

File: packages/ledgerstate/address.go [hunk 1]
```diff
@@ -0,0 +1,265 @@
...
+}
+
+// ED25519AddressFromMarshalUtil is a method that parses an ED25519Address from the given MarshalUtil.
+func ED25519AddressFromMarshalUtil(marshalUtil *marshalutil.MarshalUtil) (address *ED25519Address, err error) {
+	addressType, err := marshalUtil.ReadByte()
+	if err != nil {
+		err = xerrors.Errorf("failed to parse AddressType (%v): %w", err, ErrParseBytesFailed)
+		return
+	}
+	if AddressType(addressType) != AddressTypeBLS {
```
Comment (lzpap) [inline packages/ledgerstate/address.go:140]: Shouldn't this be `AddressTypeED25519`?
```suggestion
	if AddressType(addressType) != AddressTypeED25519{
```

File: packages/ledgerstate/address.go [hunk 2]
```diff
@@ -0,0 +1,265 @@
...
+// NewBLSAddress creates a new BLSAddress from then given public key.
+func NewBLSAddress(publicKey []byte) *BLSAddress {
+	digest := blake2b.Sum256(publicKey)
+
+	return &BLSAddress{
+		digest: digest[:],
+	}
+}
+
+// BLSAddressFromBytes unmarshals an ED25519Address from a sequence of bytes.
```
Comment (lzpap) [inline packages/ledgerstate/address.go:202]:
```suggestion
// BLSAddressFromBytes unmarshals an BLSAddress from a sequence of bytes.
```

File: packages/ledgerstate/address.go [hunk 3]
```diff
@@ -0,0 +1,265 @@
...
+// endregion ///////////////////////////////////////////////////////////////////////////////////////////////////////////
+
+// region ED25519Address ///////////////////////////////////////////////////////////////////////////////////////////////
+
+// ED25519Address represents an Address that is secured by the ED25519 signature scheme.
+type ED25519Address struct {
+	digest []byte
+}
+
+// NewED25519Address creates a new ED25519Address from then given public key.
```
Comment (jonastheis) [inline packages/ledgerstate/address.go:112]:
```suggestion
// NewED25519Address creates a new ED25519Address from the given public key.
```

File: packages/ledgerstate/address.go [hunk 4]
```diff
@@ -0,0 +1,265 @@
...
+// endregion ///////////////////////////////////////////////////////////////////////////////////////////////////////////
+
+// region BLSAddress ///////////////////////////////////////////////////////////////////////////////////////////////////
+
+// BLSAddress represents an Address that is secured by the BLS signature scheme.
+type BLSAddress struct {
+	digest []byte
+}
+
+// NewBLSAddress creates a new BLSAddress from then given public key.
```
Comment (jonastheis) [inline packages/ledgerstate/address.go:193]:
```suggestion
// NewBLSAddress creates a new BLSAddress from the given public key.
```

**Pull request merged**